In [3]:
data<-read.csv(choose.files())

In [4]:
data

Age    Gender OwnHome Married Location Salary Children History  Catalogs
1    Young  Female Rent    Married Far      30100  0        2 Medium 12      
2    Young  Female Rent    Single  Close    11700  0        1 Low    12      
3    Young  Female Rent    Single  Close    14000  0        1 Low    12      
4    Young  Female Rent    Married Close    76400  0        4 NA     12      
5    Young  Female Rent    Single  Close    15600  0        4 NA     12      
6    Young  Female Rent    Single  Close    17300  0        1 Low    12      
7    Young  Female Own     Married Close    76200  0        3 High   12      
8    Young  Female Own     Single  Far      31900  0        4 NA     12      
9    Young  Female Own     Single  Far      16300  0        4 NA     12      
10   Young  Female Rent    Married Close    61700  0        4 NA     12      
11   Young  Male   Rent    Married Close    49800  0        2 Medium 12      
12   Young  Male   Rent    Single  Close    18700  0        1 Low    12      
13   Young  Male   Rent    Single  Close    21900  0        1 Low    12      
14   Young  Male   Rent    Single  Close    15200  0        4 NA     12      
15   Young  Male   Rent    Single  Close    21400  0        4 NA     12      
16   Young  Male   Rent    Single  Close    23500  0        1 Low    12      
17   Young  Male   Own     Single  Close    33400  0        4 NA     12      
18   Young  Male   Rent    Single  Close    14100  0        1 Low    12      
19   Young  Male   Rent    Married Close    67800  0        3 High   12      
20   Young  Male   Rent    Married Far      25100  0        2 Medium 12      
21   Young  Male   Rent    Single  Close    17400  0        4 NA     12      
22   Young  Male   Own     Single  Far      39600  0        4 NA     12      
23   Young  Male   Rent    Single  Close    21200  0        4 NA     12      
24   Young  Male   Rent    Married Far      51600  0        4 NA     12      
25   Young  Male   Rent    Married Close    36700  0        2 Medium 12      
26   Young  Male   Rent    Married Far      55000  0        4 NA     12      
27   Young  Male   Rent    Single  Far      19000  0        4 NA     12      
28   Young  Male   Rent    Single  Far      25300  0        2 Medium 12      
29   Young  Male   Rent    Single  Close    19600  0        1 Low    12      
30   Young  Male   Rent    Single  Close    18000  0        1 Low    12      
...  ...    ...    ...     ...     ...      ...    ...      ...      ...     
971  Young  Male   Rent    Married Close     64300 3        1 Low    6       
972  Young  Male   Rent    Single  Close     13000 3        1 Low    6       
973  Young  Male   Rent    Single  Close     14600 3        4 NA     6       
974  Young  Male   Rent    Single  Close     15000 3        1 Low    6       
975  Middle Female Own     Married Close     79400 3        1 Low    6       
976  Middle Female Own     Married Close    130700 3        4 NA     6       
977  Middle Female Own     Single  Far       56000 3        2 Medium 6       
978  Middle Female Rent    Married Close    104900 3        1 Low    6       
979  Middle Female Own     Single  Close     46600 3        1 Low    6       
980  Middle Female Rent    Married Close     97400 3        1 Low    6       
981  Middle Female Own     Married Close     51900 3        1 Low    6       
982  Middle Female Own     Single  Far       62500 3        4 NA     6       
983  Middle Female Rent    Single  Close     41400 3        1 Low    6       
984  Middle Female Own     Single  Close     46300 3        4 NA     6       
985  Middle Female Rent    Single  Far       39000 3        1 Low    6       
986  Middle Female Own     Married Close    131500 3        4 NA     6       
987  Middle Female Rent    Married Close     72900 3        1 Low    6       
988  Middle Female Own     Single  Close     53300 3        4 NA     6       
989  Middle Female Rent    Married Close     88600 3        1 Low    6       
990  Middle Male   Rent    

In [7]:
m1<-lm(data$AmountSpent~data$Catalogs*data$Salary)
summary(m1)


Call:
lm(formula = data$AmountSpent ~ data$Catalogs * data$Salary)

Residuals:
     Min       1Q   Median       3Q      Max 
-2138.38  -247.57   -17.82   228.50  2913.79 

Coefficients:
                            Estimate Std. Error t value Pr(>|t|)    
(Intercept)                6.855e+01  8.680e+01   0.790    0.430    
data$Catalogs             -4.001e-01  5.739e+00  -0.070    0.944    
data$Salary                6.095e-03  1.461e-03   4.171  3.3e-05 ***
data$Catalogs:data$Salary  9.434e-04  9.104e-05  10.363  < 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 569.6 on 996 degrees of freedom
Multiple R-squared:  0.6498,	Adjusted R-squared:  0.6488 
F-statistic: 616.1 on 3 and 996 DF,  p-value: < 2.2e-16


In [8]:
m2<-lm(data$AmountSpent~data$Catalogs+data$Salary)
summary(m2)


Call:
lm(formula = data$AmountSpent ~ data$Catalogs + data$Salary)

Residuals:
    Min      1Q  Median      3Q     Max 
-1761.3  -327.9    14.6   270.6  3387.8 

Coefficients:
                Estimate Std. Error t value Pr(>|t|)    
(Intercept)   -6.591e+02  5.368e+01  -12.28   <2e-16 ***
data$Catalogs  5.170e+01  2.912e+00   17.75   <2e-16 ***
data$Salary    1.991e-02  6.299e-04   31.61   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 599.2 on 997 degrees of freedom
Multiple R-squared:  0.6121,	Adjusted R-squared:  0.6113 
F-statistic: 786.5 on 2 and 997 DF,  p-value: < 2.2e-16


# Answers for Q1
1) m1 is better than m2 as adjusted R-squared is more for m1.
2) Yes, Salary and Catalogs are both significant when interaction is not there, that is in model 1. If interaction term is considered, then catalogs is not significant.
3) m1: Amount spent will be decreased but not significantly. 
   m2: amount spent will increase by 51.7.

In [11]:
m3<-lm(data$AmountSpent~data$Salary*as.factor(data$Location)+data$Catalogs*as.factor(data$Location)+data$Children*as.factor(data$Location))
summary(m3)


Call:
lm(formula = data$AmountSpent ~ data$Salary * as.factor(data$Location) + 
    data$Catalogs * as.factor(data$Location) + data$Children * 
    as.factor(data$Location))

Residuals:
     Min       1Q   Median       3Q      Max 
-1348.39  -299.38   -40.29   213.38  2637.35 

Coefficients:
                                            Estimate Std. Error t value
(Intercept)                               -3.103e+02  5.536e+01  -5.605
data$Salary                                1.737e-02  6.051e-04  28.699
as.factor(data$Location)Far               -2.016e+02  1.019e+02  -1.979
data$Catalogs                              3.988e+01  2.850e+00  13.993
data$Children                             -1.917e+02  1.753e+01 -10.934
data$Salary:as.factor(data$Location)Far    1.212e-02  1.136e-03  10.669
as.factor(data$Location)Far:data$Catalogs  5.890e+00  5.297e+00   1.112
as.factor(data$Location)Far:data$Children -4.988e+01  3.237e+01  -1.541
                                          Pr(>|t|)    
(In

# Answers for Q2
Regression Equation:


a) Amount Spent = -310.3 + 0.01737 X Salary + 39.98 X Catalogs - 191.7 X Children -0.02 X Far - 0.01212 X Salary X Far + 5.89 X Far X Catalogs - 0.049 X Far X Children


b) Salary and Location interaction term are significant.

In [16]:
m4<-lm(data$AmountSpent~data$Salary*data$Catalogs+data$Catalogs*data$Children+data$Catalogs*as.factor(data$Location))
summary(m4)


Call:
lm(formula = data$AmountSpent ~ data$Salary * data$Catalogs + 
    data$Catalogs * data$Children + data$Catalogs * as.factor(data$Location))

Residuals:
     Min       1Q   Median       3Q      Max 
-1355.88  -262.14   -40.02   196.08  2260.44 

Coefficients:
                                            Estimate Std. Error t value
(Intercept)                                1.464e+02  8.344e+01   1.755
data$Salary                                7.134e-03  1.223e-03   5.832
data$Catalogs                             -6.124e+00  5.376e+00  -1.139
data$Children                             -6.154e+01  3.333e+01  -1.847
as.factor(data$Location)Far                1.151e+02  8.476e+01   1.358
data$Salary:data$Catalogs                  9.318e-04  7.603e-05  12.255
data$Catalogs:data$Children               -1.024e+01  2.124e+00  -4.820
data$Catalogs:as.factor(data$Location)Far  2.461e+01  5.030e+00   4.894
                                          Pr(>|t|)    
(Intercept)                   

# Answers for Q4

a) m4 is slightly better than m3 as adjusted R-squared value is slightly greater for m4.

b) all three interaction terms are significant as all the three p-values are less than 0.05.